fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/AggregateDisassemble.java

```diff
@@ -54,86 +56,192 @@
-            //    | situation | origin expression   | local output expression | expression in global aggregate |
-            //    +-----------+---------------------+-------------------------+--------------------------------+
-            //    | a         | Ref(k1)#1           | Ref(k1)#1               | Ref(k1)#1                      |
-            //    +-----------+---------------------+-------------------------+--------------------------------+
-            //    | b         | Ref(k1)#1 + 1       | A(Ref(k1)#1 + 1, key)#2 | Ref(key)#2                     |
-            //    +-----------+---------------------+-------------------------+--------------------------------+
-            //    | c         | A(AF(v1#1), 'af')#2 | A(AF(v1#1), 'af')#3     | AF(af#3)                       |
-            //    +-----------+---------------------+-------------------------+--------------------------------+
-            //    NOTICE: Ref: SlotReference, A: Alias, AF: AggregateFunction, #x: ExprId x
-            // 2. collect local aggregate output expressions and local aggregate group by expression list
-            Map<Expression, Expression> inputSubstitutionMap = Maps.newHashMap();
-            List<Expression> localGroupByExprs = aggregate.getGroupByExpressions();
-            List<NamedExpression> localOutputExprs = Lists.newArrayList();
-            for (Expression originGroupByExpr : originGroupByExprs) {
-                if (inputSubstitutionMap.containsKey(originGroupByExpr)) {
+    private Map<Expression, Expression> inputSubstitutionMap = Maps.newHashMap();
+    private List<NamedExpression> distinctAggFunctionParams = new ArrayList<>();
+    private List<AggregateFunction> distinctAggFunctions = new ArrayList<>();
+    private List<NamedExpression> distinctOriginOutputExprs = new ArrayList<>();
+
+    // only support distinct function with group by
+    // TODO: support distinct function without group by. (add second global phase)
+    private LogicalAggregate secondDisassemble(LogicalAggregate<GroupPlan> aggregate) {
+        // origin sql: select count(distinct a) from t1 group by b;
+        // global agg: select a from t1 group by b, a;
+        // second local agg: select count(distinct a) from t1 group by b;
+        // In order to get the second local agg from global agg:
+        // 1. the distinct expression needs to be removed from the output and the group by of global agg
+        // 2. add distinct agg function back to output
+        List<NamedExpression> originFirstGlobalOutputExprs = aggregate.getOutputExpressions();
+        List<Expression> originFirstGlobalGroupByExprs = aggregate.getGroupByExpressions();
+        List<NamedExpression> secondLocalOutputExprs = new ArrayList<>(aggregate.getOutputExpressions());
+        // add origin distinct function back
+        secondLocalOutputExprs.addAll(distinctOriginOutputExprs);
```

Comment:
secondLocalOutputExprs generated from aggregate output expressions, i think it already has origin distinct function?